waku/v2/waku_core/topics/content_topic.nim

```diff
@@ -46,41 +50,50 @@ proc init*(T: type NsContentTopic, application, version, name, encoding: string)
 proc `$`*(topic: NsContentTopic): string =
   ## Returns a string representation of a namespaced topic
   ## in the format `/<application>/<version>/<topic-name>/<encoding>`
-  "/" & topic.application & "/" & topic.version & "/" & topic.name & "/" & topic.encoding
+  ## Autosharding adds 2 prefix `/<gen#>/<bias-name>
+  "/" & topic.generation & "/" & topic.bias & "/" & topic.application & "/" & topic.version & "/" & topic.name & "/" & topic.encoding
 
 
 # Deserialization
 
 proc parse*(T: type NsContentTopic, topic: ContentTopic|string): ParsingResult[NsContentTopic] =
   ## Splits a namespaced topic string into its constituent parts.
   ## The topic string has to be in the format `/<application>/<version>/<topic-name>/<encoding>`
+  ## Autosharding adds 2 prefix `/<gen#>/<bias-name>
 
   if not topic.startsWith("/"):
     return err(ParsingError.invalidFormat("topic must start with slash"))
 
   let parts = topic[1..<topic.len].split("/")
-  if parts.len != 4:
+  if parts.len != 6:
     return err(ParsingError.invalidFormat("invalid topic structure"))
 
+  let gen = parts[0]
+  if gen.len == 0:
```

Comment:
Should we also check that gen is a number? Or is that too restrictive for cases where people want to "fork" the network and come up with their own approach?